### ISSUE TITLE
About the implementation of MI

### ISSUE THREAD

**Issue by mingcheung** [ZhengyuZhao/Targeted-Tansfer#5]
Why the momentum in CE is implemented by "grad_a = grad_c / torch.mean(torch.abs(grad_c), (1, 2, 3), keepdim=True) + 1 * grad_pre #MI", but in Po+Trip and Logit is implemented by "grad_a = grad_c + 1 * grad_pre #MI"?

**Comment by ZhengyuZhao** [comment]
The gradient normalization can alleviate (to a small extent) the gradient decrease problem of the CE loss and so yield better performance.  Po+Trip and Logit losses by design do not suffer from this gradient vanishing problem. The official implementation of the Po+Trip paper also uses this in the same way.

**Comment by ZhengyuZhao** [comment]
> Why the momentum in CE is implemented by "grad_a = grad_c / torch.mean(torch.abs(grad_c), (1, 2, 3), keepdim=True) + 1 * grad_pre #MI", but in Po+Trip and Logit is implemented by "grad_a = grad_c + 1 * grad_pre #MI"?

The gradient normalization can alleviate (to a small extent) the gradient decrease problem of the CE loss and so yield better performance. Po+Trip and Logit losses by design do not suffer from this gradient vanishing problem. The official implementation of the Po+Trip paper also uses this in the same way.
